src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs

```diff
@@ -537,21 +537,31 @@ private void RunBackupThread(PeriodicBackup periodicBackup, BackupTask backupTas
             }
             catch (Exception e)
             {
+                if (_logger.IsOperationsEnabled)
+                    _logger.Operations($"Failed to run the backup thread: '{periodicBackup.Configuration.Name}'", e);
+
                 tcs.SetException(e);
             }
             finally
             {
-                _serverStore.ConcurrentBackupsSemaphore.Release();
+                try
+                {
+                    _serverStore.ConcurrentBackupsSemaphore.Release();
 
-                periodicBackup.RunningTask = null;
-                periodicBackup.RunningBackupTaskId = null;
-                periodicBackup.CancelToken = null;
-                periodicBackup.RunningBackupStatus = null;
+                    periodicBackup.RunningTask = null;
+                    periodicBackup.RunningBackupTaskId = null;
+                    periodicBackup.CancelToken = null;
+                    periodicBackup.RunningBackupStatus = null;
 
-                if (periodicBackup.HasScheduledBackup() && _cancellationToken.IsCancellationRequested == false)
+                    if (periodicBackup.HasScheduledBackup() && _cancellationToken.IsCancellationRequested == false)
+                    {
+                        var newBackupTimer = GetTimer(periodicBackup.Configuration, periodicBackup.BackupStatus);
+                        periodicBackup.UpdateTimer(newBackupTimer, discardIfDisabled: true);
+                    }
+                }
+                catch (Exception e)
                 {
-                    var newBackupTimer = GetTimer(periodicBackup.Configuration, periodicBackup.BackupStatus);
-                    periodicBackup.UpdateTimer(newBackupTimer, discardIfDisabled: true);
+                    tcs.SetException(e);
```

Comment:
Need to log this error.
You cannot call SetException after calling SetResult (at this point the backup task has already completed.

Reply:
I think we need to 'gather' all previous catches, execute finally and only after success of finally, set appropriate state?